share/spack/gitlab/cloud_pipelines/stacks/radiuss/spack.yaml

```diff
@@ -0,0 +1,128 @@
+# This file is currently being customized for quartz/toss_3_x86_64_ib.
+
+spack:
+  concretization: separately
+  view: false
+  config:
+    concretizer: clingo
+    install_tree:
+      root: /home/software/radiuss
+      padded_length: 512
+      projections:
+        all: '{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash}'
+
+  packages::
+    all:
+      permissions:
+        read: world
+      providers:
+        mpi: [mvapich2]   # white238
+        #mpi: [mvapich2, openmpi]  # Is this why craylib-sci shows up?
+        # Spack defaults (since limiting the scope)
+        D: [ldc]
+        awk: [gawk]
+        blas: [openblas, amdblis]
+        daal: [intel-daal]
+        elf: [elfutils]
+        fftw-api: [fftw, amdfftw]
+        flame: [libflame, amdlibflame]
+        gl: [mesa+opengl, mesa18, opengl]
+        glx: [mesa+glx, mesa18+glx, opengl]
+        glu: [mesa-glu, openglu]
+        golang: [gcc]
+        iconv: [libiconv]
+        ipp: [intel-ipp]
+        java: [openjdk, jdk, ibm-java]
+        jpeg: [libjpeg-turbo, libjpeg]
+        lapack: [openblas, amdlibflame]
+        mariadb-client: [mariadb-c-client, mariadb]
+        mkl: [intel-mkl]
+        mpe: [mpe2]
+        #mpi: [openmpi, mpich]
+        mysql-client: [mysql, mariadb-c-client]
+        opencl: [pocl]
+        osmesa: [mesa+osmesa, mesa18+osmesa]
+        pil: [py-pillow]
+        pkgconfig: [pkgconf, pkg-config]
+        rpc: [libtirpc]
+        scalapack: [netlib-scalapack, amdscalapack]
+        sycl: [hipsycl]
+        szip: [libszip, libaec]
+        tbb: [intel-tbb]
+        unwind: [libunwind]
+        yacc: [bison, byacc]
+        uuid: [util-linux-uuid, libuuid]
+  definitions:
+    #- compilers: ['%gcc@8.3.1', '%clang@10.0.0']
+    - compilers: ['%gcc@8.3.1']
+    
+    # Note skipping spot since no spack package for it
+    - radiuss:
+      - ascent  # ^conduit@0.6.0
+      - axom
+      - blt
+      - caliper
+      - care
+      - chai
+      - conduit  # ^hdf5+shared
+      - flux-core
+      - flux-sched
+      - glvis   # ^mesa-glu@9.0.0 ^mesa18~llvm
+      - hypre
+      - lbann
+      - lvarray # ^raja~openmp  # per Ben Corbett Issue #23192
+      - mfem
+      - py-hatchet
+      - py-maestrowf
+      - py-merlin
+      - py-shroud
+      - raja
+      - samrai
+      - scr
+      - sundials
+      - umpire ~openmp
+      - visit   # ^mesa-glu@9.0.0
+      - xbraid
+      - zfp
+
+  mirrors:
+    mirror: "s3://spack-binaries-develop/radiuss"
+
+  specs:
+  - matrix:
+    - [$radiuss]
+    - [$compilers]
+
+  gitlab-ci:
+    image: { "name": "ghcr.io/scottwittenburg/ecpe4s-ubuntu18.04-runner-x86_64:2020-09-01", "entrypoint": [""] }
+    script:
+      - . "./share/spack/setup-env.sh"
+      - spack --version
+      - cd share/spack/gitlab/cloud_pipelines/stacks/radiuss
```

Comment:
When you rebase on #22887 (or I guess on `develop` after that's merged) this will change to the concrete environment directory exported as an artifact by the pipeline generation job.  

So you would change  this line to:

```
cd ${SPACK_CONCRETE_ENV_DIR}
```

You don't have to do anything on the pipeline generation side, that's handled in the base `.generate` job.

Reply:
Sorry it's so unclear.  The jobs that do the rebuilds *must* change into the directory where the concrete environment exists.  You could specify that path explicitly here if you know what it is, but we provide `$SPACK_CONCRETE_ENV_DIR` for you if not.    The so-called "service jobs" (rebuild-index, clean up, nothing to rebuild) do not need it (but I'll fix it so we provide the concrete environment artifacts to those jobs too in case it ever turns out to be useful).  Your most recent pipeline failed because the rebuild jobs are changing into `share/spack/gitlab/cloud_pipelines/stacks/radiuss` directory, and there's no `spack.lock` there.   To summarize, in custom scripts for rebuild jobs, change into the `$SPACK_CONCRETE_ENV_DIR`, for the `service-job-attributes`, change into `share/spack/gitlab/cloud_pipelines/stacks/radiuss`.

Reply:
Looks like I didn't read your comment closely enough. Thanks for the clarification.